File: 프로그래머스/3/42884. 단속카메라/단속카메라.java
import java.util.*;

class Solution {
    public int solution(int[][] routes) {
        
        Arrays.sort(routes, new Comparator<int[]>() {
            @Override
            public int compare(int[] a, int[] b) {
                return Integer.compare(a[1], b[1]);
            }
        });
        
        int answer = 1;
        int end = routes[0][1]; 

        for (int i = 1; i < routes.length; i++) {
            if (routes[i][0] <= end) {
                continue;
            }

            answer++;
            end = routes[i][1];
        }
        
        return answer;
    }
}

File: 프로그래머스/3/42884. 단속카메라/단속카메라.cpp
#include <string>
#include <vector>
#include <bits/stdc++.h>

using namespace std;

int solution(vector<vector<int>> routes) {
    int answer = 1;
    
    sort(routes.begin(), routes.end(), [](const vector<int>& a, const vector<int>& b) {
        return a[1] < b[1];
    });
    
    int end = routes[0][1];
    for(int i=1; i<routes.size(); i++){
        if (routes[i][0] <= end) continue;
        else {
            answer++;
            end = routes[i][1];
        }
    }
    
    return answer;
}